base/show.jl

```diff
@@ -802,10 +802,18 @@ function show_unquoted(io::IO, ex::Expr, indent::Int, prec::Int)
             if (na == 2 || (na > 2 && func in (:+, :++, :*))) &&
                     all(!isa(a, Expr) || a.head !== :... for a in func_args)
                 sep = " $func "
+                # parenthesize (-x) ^ y for negative x
```

Comment:
That would make `:(-a + -b)` print as `:((-a) + -b)` which is quite strange. It should either be `:(-a + -b)` or `:((-a) + (-b))`, and in that case, it should also definitely extend to `:((-a) ^ (-b))`. But that is really ugly, IMO. I think the most important thing is that the expressions print and parse correctly, which as far as I have found, is only broken by `^` here. But I guess for user-defined infix operators, it would break for any operator which doesn't just pass-through on the negation with similar or higher precedence.